app/code/core/Mage/Catalog/Model/Category/Api.php

```diff
@@ -65,7 +65,7 @@ public function level($website = null, $store = null, $categoryId = null)
                         $ids = $categoryId;
                     }
                 } elseif (in_array($store, $website->getStoreIds())) {
-                    $storeId = Mage::app()->getStore($store)->getId();
+                    $store = Mage::app()->getStore($store);
```

Comment:
From the code I would expect this to cause propblems, as `$storeId` is initially set with DEFAULT_STORE_ID, and then used in line 100 after the if,elseif,else in 
```
 $collection = Mage::getModel('catalog/category')->getCollection()
            ->setStoreId($storeId)	            ->setStoreId($storeId)
            ->addAttributeToSelect('name')	            ->addAttributeToSelect('name')
            ->addAttributeToSelect('is_active');	            ->addAttributeToSelect('is_active');
```

Reply:
@Flyingmana thanks for review. I've tested it, but I'm not sure I've covered eventualities. I'll update later.

Reply:
Suggestion:

```suggestion
                    $store = Mage::app()->getStore($store);
                    $storeId = $store->getId();
```